File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/VarDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.Var;

import java.util.List;

@Dao
public interface VarDao {

    @Query("SELECT * FROM vars ORDER BY id")
    LiveData<List<Var>> getVars();

    @Query("SELECT * FROM vars WHERE id = :varId")
    LiveData<Var> getVarById(int varId);

    @Insert
    void insertVar(Var var);

    @Delete
    void deleteVar(Var var);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateVar(Var var);

}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/VarDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.Var

@Dao
interface VarDao {
    @Query("SELECT * FROM vars ORDER BY id")
    fun getVars(): LiveData<List<Var>>

    @Query("SELECT * FROM vars WHERE id = :varId")
    fun getVarById(varId: Int): LiveData<Var>

    @Insert
    fun insertVar(`var`: Var)

    @Delete
    fun deleteVar(`var`: Var)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateVar(`var`: Var)
}